options/options_test.go

```diff
@@ -25,6 +27,55 @@ const (
 	ShouldFail
 )
 
+// TOOLS-2287 specifies the desired behavior in the case that
+// the user specifies unsupported URI options.
+// This test case ensures the proper logging behavior.
+func TestLogUnsupportedOptions(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.UnitTestType)
+	
+	type TestCase struct {
+		description              string
+		args, unsupportedOptions []string
+	}
+
+	// Test matrix for logging behavior
+	cases := []TestCase{
+		{
+			description: "No unsupported options",
+			args:        strings.Split("--uri mongodb://mongodb.test.com:27017", " "),
+		},
+		{
+			description:        "Detect unsupported options",
+			args:               strings.Split("--uri mongodb://mongodb.test.com:27017/?foo=bar", " "),
+			unsupportedOptions: []string{"foo"},
```

Comment:
This could just be a string rather than a slice.

Reply:
It would need to at least be an array, because `ToolOptions.ParseArgs(...)` needs an array as input, and this field is intended to populate that function. Therefore, I figured it'd be a good idea to store it as an array to begin with.

With that said, I can omit the `--uri` flag and turn it into a statically-defined array instead of a slice, which I think would be better.